iterable-extension/src/main/java/com/mparticle/ext/iterable/IterableExtension.java

```diff
@@ -589,6 +590,35 @@ public void processCustomEvent(CustomEvent event) throws IOException {
         }
     }
 
+    @Override
+    public void processScreenViewEvent(ScreenViewEvent event) throws IOException {
+        if (processSubscribeEvent(event)) {
```

Comment:
@samdozor I'm trying to understand the semantics of `processSubscribeEvent` and when it returns `true`/`false`. I noticed Ho Joon wrote the `processScreenViewEvent` which seem to be resemble the `processCustomEvent` method.

We are forming an API/integrations team who will own this integration, but in the meantime, I, being on the infrastructure side am going to be helping our team merge PRs. It may be a while since you last touched this code, thanks for your help!

Reply:
hey @ktham processSubscribeEvent specifically checks if the event name matches the `UPDATE_SUBSCRIPTIONS_CUSTOM_EVENT_NAME` constant. If so, it treats the event specially as a subscribe event, otherwise it returns false and the event is handled generically by `processScreenViewEvent`